hazelcast/src/main/java/com/hazelcast/client/impl/protocol/template/ResponseTemplate.java

```diff
@@ -160,11 +160,17 @@ void Exception(int errorCode, String className, @Nullable String message, StackT
             , int causeErrorCode, @Nullable String causeClassName);
 
     /**
-     * @param readCount Number of items in the response.
+     * @param readCount The number of items read from the ringbuffer. This can be different from
+     *                  the size of the item array if a filter was applied when reading and some
+     *                  items were skipped and are not in the returned array. This can also be
+     *                  seen as the difference between the sequence ID of the last and first
+     *                  item in the returned array and the delta with which the user should
+     *                  increase his sequence counter to request the next set from the ringbuffer.
      * @param items     The array of serialized items.
+     * @param itemSeqs  sequence IDs of returned ringbuffer items
      */
     @Response(ResponseMessageConst.READ_RESULT_SET)
-    void ReadResultSet(int readCount, List<Data> items);
+    void ReadResultSet(int readCount, List<Data> items, @Since("1.5") @Nullable long[] itemSeqs);
```

Comment:
This PR is a preparation for the event journal PR : https://github.com/hazelcast/hazelcast-client-protocol/pull/88. I can even merge the changes into one PR - the split was made to make reviews on hazelcast easier.
In the event journal PR the map and cache templates use this response message and they need the sequence IDs. Also, ringbuffer users will need the sequence IDs once we add filtering to the reliable topic subscribers. Without the sequences for each item, the relablie topic subscriber or event journal reader cannot replay items from a certain point - it only knows the sequence ID of the first and last item.
Should I describe this in the javadoc or is there some change that I should make to this response message?